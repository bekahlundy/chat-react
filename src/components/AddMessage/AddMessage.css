.AddMessage--container {
  bottom: 0;
  position: absolute;
  text-align: center;
  width: 100%;
}

.AddMessage--input {
  width: 85%;
}
