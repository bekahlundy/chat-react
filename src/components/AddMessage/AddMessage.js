import React, { Component } from "react";
import "./AddMessage.css";

export class AddMessage extends Component {
  constructor(props) {
    super(props);
    this.state = {
      message: ""
    };
  }

  handleKeyUp = ev => {
    if (ev.keyCode === 13) {
      this.handleClick();
    }
  };

  handleClick = () => {
    if (this.state.message) {
      this.props.sendMessage(this.state.message);
      this.setState({ message: "" });
    }
  };

  render() {
    const { message } = this.state;

    return (
      <div className="AddMessage--container">
        <input
          type="text"
          data-testid="add-message-input"
          placeholder="Message"
          className="AddMessage--input"
          value={message}
          onChange={ev => this.setState({ message: ev.target.value })}
          onKeyUp={this.handleKeyUp}
        />
        <button
          data-testid="add-message-button"
          onClick={this.handleClick}
          className="AddMessage--btn"
        >
          Send
        </button>
      </div>
    );
  }
}
