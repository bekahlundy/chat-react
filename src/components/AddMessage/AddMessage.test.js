import React from "react";
import ReactDOM from "react-dom";
import { AddMessage } from "./AddMessage";
import { render, cleanup, getByTestId, fireEvent, wait } from "react-testing-library";
import uuid from 'uuid';

describe("AddMessage", () => {

  afterEach(() => {
    cleanup();
  });

  it("renders", () => {
    const sendMessageMock = jest.fn();

    const { container } = render(<AddMessage sendMessage={sendMessageMock} />);
    expect(container.firstChild).toMatchSnapshot();
  });

  it("calls sendMessage on submit(click) if given a message", async () => {
    const sendMessageMock = jest.fn();
    const { container } = render(<AddMessage sendMessage={sendMessageMock} />);

    const input = getByTestId(container, "add-message-input");
    fireEvent.change(input, { target: { value: "Hello hello"} } );

    const button = getByTestId(container, "add-message-button");
    fireEvent.click(button);

    await wait(() => {
      expect(sendMessageMock).toHaveBeenCalledWith('Hello hello');
    });
  });

  xit("calls sendMessage on submit(enter) if given a message", async () => {
    const sendMessageMock = jest.fn();
    const { container } = render(<AddMessage sendMessage={sendMessageMock} />);

    const input = getByTestId(container, "add-message-input");
    fireEvent.change(input, { target: { value: "Hello hello"} } );
    fireEvent.keyUp(input, { key: 'Enter', code: 13, charCode: 13 });

    await wait(() => {
      expect(sendMessageMock).toHaveBeenCalledWith('Hello hello');
    });
  });

  it("does not call sendMessage on submit(click) when not given a message", async () => {
    const sendMessageMock = jest.fn();
    const { container } = render(<AddMessage sendMessage={sendMessageMock} />);

    const button = getByTestId(container, "add-message-button");
    fireEvent.click(button);

    await wait(() => {
      expect(sendMessageMock).not.toHaveBeenCalled();
    });
  });

  xit("does not call sendMessage on submit(enter) when not given a user", async () => {
    const sendMessageMock = jest.fn();
    const { container } = render(<AddMessage sendMessage={sendMessageMock} />);

    const input = getByTestId(container, "add-message-input");
    fireEvent.keyUp(input, { key: 'Enter', code: 13, charCode: 13 });

    await wait(() => {
      expect(sendMessageMock).not.toHaveBeenCalled();
    });
  });
});
