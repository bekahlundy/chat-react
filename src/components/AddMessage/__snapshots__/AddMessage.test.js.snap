// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`AddMessage renders 1`] = `
<div
  class="AddMessage--container"
>
  <input
    class="AddMessage--input"
    data-testid="add-message-input"
    placeholder="Message"
    type="text"
    value=""
  />
  <button
    class="AddMessage--btn"
    data-testid="add-message-button"
  >
    Send
  </button>
</div>
`;
